libraries/rush-lib/src/logic/operations/BuildTimePlugin.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { IExecutionResult } from './IOperationExecutionResult';
+import { OperationStatus } from './OperationStatus';
+
+const PLUGIN_NAME: 'BuildTimePlugin' = 'BuildTimePlugin';
+
+interface ITimelineRecord {
+  startTime: number;
+  endTime: number;
+  durationString: string;
+  name: string;
+  status: OperationStatus;
+}
+
+export interface IBuildTimeRecord {
+  project: string;
+  buildTime: number;
+  status: OperationStatus;
+}
+
+export class BuildTimePlugin implements IPhasedCommandPlugin {
+  private _buildTimes: IBuildTimeRecord[];
+
+  public constructor() {
+    this._buildTimes = [];
+  }
+
+  public apply(hooks: PhasedCommandHooks): IBuildTimeRecord[] {
+    hooks.afterExecuteOperations.tap(
+      PLUGIN_NAME,
+      (result: IExecutionResult, context: ICreateOperationsContext): void => {
+        this._buildTimes = _setBuildTimes(result);
+      }
+    );
+    return this._buildTimes;
+  }
+}
+
+export function _setBuildTimes(result: IExecutionResult): IBuildTimeRecord[] {
+  const data: ITimelineRecord[] = [];
+  const buildTimes: IBuildTimeRecord[] = [];
+
+  for (const [operation, operationResult] of result.operationResults) {
+    if (operation.runner?.silent) {
+      continue;
+    }
+
+    const { stopwatch } = operationResult;
+
+    const { startTime, endTime } = stopwatch;
+
+    if (startTime && endTime) {
+      const { duration } = stopwatch;
+      const durationString: string = duration.toFixed(2);
+
+      data.push({
+        startTime,
+        endTime,
+        durationString,
+        name: operation.name!.split(' ')[0],
```

Comment:
What is this attempting to extract from the operation name? If it's the project name, that can be had from `operation.associatedProject.packageName`.

Reply:
the packageName is what I was trying to extract. Thanks for pointing me to this property